src/jit/lsra.cpp

```diff
@@ -5823,6 +5823,10 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
             assert((bestScore & VALUE_AVAILABLE) == 0);
         }
         assignPhysReg(availablePhysRegInterval, currentInterval);
+#ifdef _TARGET_ARM_
+        if (currentInterval->registerType == TYP_DOUBLE && !genIsValidDoubleReg(availablePhysRegInterval->regNum))
```

Comment:
I need to understand why we would allocate a register that is not a valid double register.

Reply:
I also think so, and it should be an assertion instead.
For example, 
```
assert( currentInterval->registerType != TYP_DOUBLE
           || (currentInterval->registerType == TYP_DOUBLE && genIsValidDoubleReg(availablePhysRegInterval->regNum))
```

Reply:
OK. I can check it. Thank you.